util/godfig/src/backend/mod.rs

```diff
@@ -53,4 +56,11 @@ pub trait BackendOperations {
         T: serde::de::DeserializeOwned + serde::Serialize + Send,
         F: FnOnce(Option<T>) -> Fut + Send,
         Fut: std::future::Future<Output = Result<Option<T>, GodfigBackendError>> + Send;
+
+    async fn try_transaction_with_result<K, T, R, F, Fut>(&self, key: K, callback: F) -> Result<R, GodfigBackendError>
```

Comment:
Can this be documented? There are five template parameters, so it's a bit tricky to decipher. What are the expectations for the closure? It works on `Option<T>` and returns `(Option<T>, R)`, I guess as a way to insert or remove or change the value and return some result.

The `try_` prefix is not necessary.

The `try_transaction` method (or, better, `transaction`) can now have a default implementation in terms of `try_transaction_with_result`.